bundles/org.openhab.binding.wled/src/main/java/org/openhab/binding/wled/internal/handlers/WLedSegmentHandler.java

```diff
@@ -216,105 +227,35 @@ public void handleCommand(ChannelUID channelUID, Command command) {
-                        }
-                    }
-                    break;
-                case CHANNEL_PRESET_CYCLE: // ch removed in firmware 0.13.0 and newer
-                    if (command instanceof OnOffType) {
-                        localApi.setPresetCycle(OnOffType.ON.equals(command));
-                    }
-                    break;
             }
         } catch (ApiException e) {
             logger.debug("Exception occured:{}", e.getMessage());
         }
     }
 
-    public void savePreset(int position, String presetName) {
-        try {
-            if (api != null) {
-                api.savePreset(position, presetName);
-            }
-        } catch (ApiException e) {
-        }
-    }
-
-    public void removeChannels(ArrayList<Channel> removeChannels) {
-        if (!removeChannels.isEmpty()) {
-            ThingBuilder thingBuilder = editThing();
-            thingBuilder.withoutChannels(removeChannels);
-            updateThing(thingBuilder.build());
-        }
-    }
-
-    public void update(String channelID, State state) {
-        updateState(channelID, state);
-    }
-
-    private void pollState() {
-        WledApi localApi = api;
-        try {
-            if (localApi == null) {
-                api = apiFactory.getApi(this);
-                api.initialize();
-            }
-            if (localApi == null) {
-                return;
-            }
-            localApi.update();
-            updateStatus(ThingStatus.ONLINE);
-        } catch (ApiException e) {
-            api = null;// Firmware may be updated so need to check next connect
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-        }
-    }
-
     @Override
     public void initialize() {
-        config = getConfigAs(WLedConfiguration.class);
-        if (config.segmentIndex < 0) {
-            config.segmentIndex = 0;
-        }
-        if (!config.address.contains("://")) {
-            logger.debug("Address was not entered in correct format, it may be the raw IP so adding http:// to start");
-            config.address = "http://" + config.address;
-        }
-        pollingFuture = scheduler.scheduleWithFixedDelay(this::pollState, 0, config.pollTime, TimeUnit.SECONDS);
-    }
-
-    @Override
-    public void dispose() {
-        Future<?> future = pollingFuture;
-        if (future != null) {
-            future.cancel(true);
-            pollingFuture = null;
+        config = getConfigAs(WLedSegmentConfiguration.class);
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "No bridge is selected.");
+        } else {
+            WLedBridgeHandler localBridgeHandler = (WLedBridgeHandler) bridge.getHandler();
+            if (localBridgeHandler == null) {
+                return;
```

Comment:
Always set a thing status in the initialize() method, so:
updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALED);